pinot-query-runtime/src/test/resources/queries/Comparisons.json

```diff
@@ -0,0 +1,516 @@
+      }
+    },
+    "queries": [
+      { "sql": "SELECT i1 < i2, i2 < i1 FROM {tbl}" },
+      { "sql": "SELECT i1 <= i2, i2 <= i1 FROM {tbl}" },
+      { "sql": "SELECT i1 > i2, i2 > i1 FROM {tbl}" },
+      { "sql": "SELECT i1 >= i2, i2 >= i1 FROM {tbl}" },
+      { "sql": "SELECT i1 != i2, i2 != i1 FROM {tbl}" },
+      { "sql": "SELECT i1 <> i2, i2 <> i1 FROM {tbl}" },
+      { "sql": "SELECT i1 = i2, i2 = i1 FROM {tbl}" },
+      { "sql": "SELECT i1 IS DISTINCT FROM i2, i2 IS DISTINCT FROM i1 FROM {tbl}" }
+    ]
+  },
+  "int_x_numeric_comp": {
+    "psql": "9.2",
+    "tables": {
+      "tbl": {
+        "schema": [
+          {"name": "i1", "type": "INT"},
+          {"name": "i2", "type": "BIG_DECIMAL"}
+        ],
+        "inputs": [
+          [1, "2.0"], [1, "0.9"], [-1, "-2.0"], [0, "0.0"], [2147483647, "-2147483648"]
+        ]
+      }
+    },
+    "queries": [
+      { "sql": "SELECT i1 < i2, i2 < i1 FROM {tbl}" },
+      { "sql": "SELECT i1 <= i2, i2 <= i1 FROM {tbl}" },
+      { "sql": "SELECT i1 > i2, i2 > i1 FROM {tbl}" },
+      { "sql": "SELECT i1 >= i2, i2 >= i1 FROM {tbl}" },
+      { "sql": "SELECT i1 != i2, i2 != i1 FROM {tbl}" },
+      { "sql": "SELECT i1 <> i2, i2 <> i1 FROM {tbl}" },
+      { "sql": "SELECT i1 = i2, i2 = i1 FROM {tbl}" },
+      { "sql": "SELECT i1 IS DISTINCT FROM i2, i2 IS DISTINCT FROM i1 FROM {tbl}" }
+    ]
+  },
+  "float_x_bigint_comp": {
+    "psql": "9.2",
+    "tables": {
+      "tbl": {
+        "schema": [
+          {"name": "i1", "type": "FLOAT"},
+          {"name": "i2", "type": "LONG"}
+        ],
+        "inputs": [
+          [1.1, 1], [0.01, 1], [1.1, 1], [-1.1, -1], [0.0, 0], [1E-37, 137]
+        ]
+      }
+    },
+    "queries": [
+      { "sql": "SELECT i1 < i2, i2 < i1 FROM {tbl}" },
+      { "sql": "SELECT i1 <= i2, i2 <= i1 FROM {tbl}" },
+      { "sql": "SELECT i1 > i2, i2 > i1 FROM {tbl}" },
+      { "sql": "SELECT i1 >= i2, i2 >= i1 FROM {tbl}" },
+      { "sql": "SELECT i1 != i2, i2 != i1 FROM {tbl}" },
+      { "sql": "SELECT i1 <> i2, i2 <> i1 FROM {tbl}" },
+      { "sql": "SELECT i1 = i2, i2 = i1 FROM {tbl}" },
+      { "sql": "SELECT i1 IS DISTINCT FROM i2, i2 IS DISTINCT FROM i1 FROM {tbl}" }
+    ]
+  },
+  "float_x_double_comp": {
+    "psql": "9.2",
+    "tables": {
+      "tbl": {
+        "schema": [
+          {"name": "i1", "type": "FLOAT"},
+          {"name": "i2", "type": "DOUBLE"}
+        ],
+        "inputs": [
+          [1.1, 1.2],
+          [1.01, 1.1],
+          [1.1, 1.1],
+          [-1.1, -1.2],
+          [0.0, 0.0],
+          [1E-37, 1E+37]
+        ]
+      }
+    },
+    "queries": [
+      { "sql": "SELECT i1 < i2, i2 < i1 FROM {tbl}" },
+      { "sql": "SELECT i1 <= i2, i2 <= i1 FROM {tbl}" },
+      { "sql": "SELECT i1 > i2, i2 > i1 FROM {tbl}" },
+      { "sql": "SELECT i1 >= i2, i2 >= i1 FROM {tbl}" },
+      { "sql": "SELECT i1 IS DISTINCT FROM i2, i2 IS DISTINCT FROM i1 FROM {tbl}" },
+      {
+        "ignored": true,
```

Comment:
this might be related to the type hoisting PR. i will test it and see if it fixes this